src/Dependencies.jl

```diff
@@ -86,14 +94,24 @@ The optional keyword argument `platforms` is a vector of `AbstractPlatform`s
 which indicates for which platforms the dependency should be used.  By default
 `platforms=[AnyPlatform()]`, to mean that the dependency is compatible with all
 platforms.
+
+The optional keyword argument `top_level` denotates that this dependency is
+platform independent. It implies that the `platforms` keyword argument is set
+to `[AnyPlatform()]`. The primary use-case is for packages that hold information
+about the platform selection using `Preferences`. Platform selection is cached
+and in the case that no platform is available we need to be able to invalidate
+said cache. Invalidation occurs through the package that owns the `Preferences`
+data.
 """
 struct Dependency <: AbstractDependency
     pkg::PkgSpec
     build_version::Union{VersionNumber,Nothing}
     compat::String  # semver string for use in Project.toml of the JLL
     platforms::Vector{<:AbstractPlatform}
+    top_level::Bool
```

Comment:
It just occurred to me that top-level dependencies can only be `RuntimeDependency`, right?  Or do you see cases where it can be a built-time dependency too?

Reply:
This would make #278 largely unnecessary because those dependencies would be pulled in during the build, and so they wouldn't cause the extra warning in the first place.

Reply:
Uhm the only case is @maleadt current adventure where the availability of a artifact might change the platform selection of the current one.

Reply:
Maybe a more principled stance would be to.make everything that is AnyPlatform top level?

Reply:
> Maybe a more principled stance would be to.make everything that is AnyPlatform top level?

Uhm, no [`iso_codes_jll`](https://github.com/JuliaBinaryWrappers/iso_codes_jll.jl) is a regular JLL which is `AnyPlatform`, I'm not sure that should be a top-level dependency.

Reply:
Also, `AnyPlatform()` packages are often build-time dependencies (e.g. header-only packages), `iso_codes_jll` is an exception but we have to deal with that.